tests/farmer_harvester/test_farmer.py

```diff
@@ -0,0 +1,519 @@
+        ),
+        pytest.param(
+            NewProofOfSpaceCase.create_verified_quality_case(
+                difficulty=uint64(1),
+                sub_slot_iters=uint64(1000000000000),
+                pool_url="localhost",
+                pool_difficulty=uint64(1),
+                authentication_token_timeout=None,
+                use_invalid_peer_response=False,
+                expected_pool_stats={
+                    "points_found_since_start": 0,
+                    # Original item format here is (timestamp, value) but we'll ignore timestamp part
+                    # so every `xxx_24h` item in this dict will be List[Any].
+                    "points_found_24h": [],
+                    "points_acknowledged_since_start": 0,
+                    "points_acknowledged_24h": [],
+                    "pool_errors_24h": [],
+                    "valid_partials_since_start": 0,
+                    "valid_partials_24h": [],
+                    "invalid_partials_since_start": 0,
+                    "invalid_partials_24h": [],
+                    "stale_partials_since_start": 0,
+                    "stale_partials_24h": [],
+                    "missing_partials_since_start": 1,
+                    "missing_partials_24h": [1],
+                },
+            ),
+            id="empty_auth_timeout",
+        ),
+        pytest.param(
+            NewProofOfSpaceCase.create_verified_quality_case(
+                difficulty=uint64(1),
+                sub_slot_iters=uint64(1000000000000),
+                pool_url="localhost",
+                pool_difficulty=uint64(1),
+                authentication_token_timeout=uint8(10),
+                use_invalid_peer_response=True,
+                expected_pool_stats={
+                    "points_found_since_start": 0,
+                    # Original item format here is (timestamp, value) but we'll ignore timestamp part
+                    # so every `xxx_24h` item in this dict will be List[Any].
+                    "points_found_24h": [],
+                    "points_acknowledged_since_start": 0,
+                    "points_acknowledged_24h": [],
+                    "pool_errors_24h": [],
+                    "valid_partials_since_start": 0,
+                    "valid_partials_24h": [],
+                    "invalid_partials_since_start": 1,
+                    "invalid_partials_24h": [1],
+                    "stale_partials_since_start": 0,
+                    "stale_partials_24h": [],
+                    "missing_partials_since_start": 0,
+                    "missing_partials_24h": [],
+                },
+            ),
+            id="invalid_peer_response",
+        ),
+    ],
+)
+@pytest.mark.asyncio
+async def test_farmer_new_proof_of_space_for_pool_stats(
+    harvester_farmer_environment: HarvesterFarmerEnvironment,
+    case: NewProofOfSpaceCase,
+) -> None:
+    farmer_service, farmer_rpc_client, _, _, _ = harvester_farmer_environment
```

Comment:
this looks like a pretty fragile test, but it's hard to tell. This is such a large test that it's hard to read it and feel confident that it's correct.

Now that you have it, it probably makes sense to keep it and simplify it to make it helpful.

However, the new function you added, `increment_pool_stats()` is very simple function, and it can (most likely) have a very simple unit test. By unit test I mean *just* setting up the values to pass in to that function and test that it does the thing you would expect.

Reply:
Resolving comments in test code - we should iterate and simplfy post release